### PR TITLE
Local and off root config for plugins

### PR DESCRIPTION
**Pull request by nxtum**
Plugins with the root() function would not work in a configuration with: (above configuration) root: Local, and (configuration) root: off  as the libraries would not be loaded. By using string iterator, it allows them to work in that config

### CODE REVIEW

File: Source/Thunder/PluginServer.h [hunk 1]
```diff
@@ -1296,8 +1296,8 @@ namespace PluginHost {
             uint32_t WakeupChildren(const Core::process_t parentPID, const uint32_t timeout);
             #endif
 
-            std::vector<string> GetLibrarySearchPaths(const string& locator) const override
-            {
+            RPC::IStringIterator* GetLibrarySearchPaths(const string& locator) const override 
+            {
```
Comment (VeithMetro) [inline Source/Thunder/PluginServer.h]: A few unnecessary spaces here and one line above 😄

File: Source/Thunder/PluginServer.h [hunk 2]
```diff
@@ -1357,9 +1356,9 @@ namespace PluginHost {
                         // the dlopen has a process wide system lock, make sure that the, during open used lock of the
                         // ServiceAdministrator, is already taken before entering the dlopen. This can only be achieved
                         // by forwarding this call to the ServiceAdministrator, so please so...
-                        Core::Library newLib = Core::ServiceAdministrator::Instance().LoadLibrary(iter->c_str());
+                        Core::Library newLib = Core::ServiceAdministrator::Instance().LoadLibrary(element.c_str());
 
-                        if (newLib.IsLoaded() == true) {
+                         if (newLib.IsLoaded() == true) {
```
Comment (VeithMetro) [inline Source/Thunder/PluginServer.h]: One more unnecessary space here 😝

File: Source/plugins/Shell.cpp [hunk 3]
```diff
@@ -41,22 +41,22 @@ namespace PluginHost
             if (locator.empty() == true) {
                 result = Core::ServiceAdministrator::Instance().Instantiate(Core::Library(), className.c_str(), version, interface);
             } else {
-                std::vector<string> all_paths = GetLibrarySearchPaths(locator);
-                std::vector<string>::const_iterator index = all_paths.begin();
-                while ((result == nullptr) && (index != all_paths.end())) {
-                    Core::File file(index->c_str());
+                RPC::IStringIterator* all_paths = GetLibrarySearchPaths(locator);
+                string element;
+                while (all_paths->Next(element) == true){
```
Comment (VeithMetro) [inline Source/plugins/Shell.cpp]: Add a space before curly bracket

File: Source/Thunder/PluginServer.h [hunk 4]
```diff
@@ -1335,19 +1335,18 @@ namespace PluginHost {
...
 
-                std::vector<string> all_paths = GetLibrarySearchPaths(name);
-                std::vector<string>::const_iterator iter = std::begin(all_paths);
-
-                while ( (iter != std::end(all_paths)) && (progressedState <= 2) ) {
-                    Core::File libraryToLoad(*iter);
-
+                RPC::IStringIterator* all_paths = GetLibrarySearchPaths(name);
+                string element;
+                while((all_paths->Next(element) == true) && (progressedState <= 2)){
```
Comment (VeithMetro) [inline Source/Thunder/PluginServer.h]: Add space before a curly bracket

**Review by VeithMetro — CHANGES_REQUESTED** [review]
Some minor formatting changes 😀

**Review by pwielders — APPROVED** [review]

**Review by VeithMetro — APPROVED** [review]

**Pull request merged**